beets/importer.py

```diff
@@ -754,6 +754,8 @@ def add(self, lib):
             self.record_replaced(lib)
             self.remove_replaced(lib)
             self.album = lib.add_album(self.imported_items())
+            if 'data_source' in self.imported_items()[0].keys():
```

Comment:
I think a simpler expression `'data_source' in self.imported_items()[0]` should work… any chance you could give that a quick try?

Reply:
No worries, all done
...I must admit I had wondered if it was possible that way, but with my inexperience and in my haste I neglected to test it first. Thanks again  :)